src/Compilers/CSharp/Test/Symbol/Symbols/CovariantReturnTests.cs

```diff
@@ -4436,5 +4456,305 @@ void verify(CSharpCompilation compilation)
+    public virtual void M(int y) => throw null; // 2
+}
+
+class Derived : Base
+{
+    public override void M(int z) => throw null; // 3
+}
+";
+            var comp = CreateCompilation(withCovariantReturns: withCovariantReturns, source);
+            comp.VerifyDiagnostics(
+                // (5,25): error CS0111: Type 'Base' already defines a member called 'M' with the same parameter types
+                //     public virtual void M(int y) => throw null; // 2
+                Diagnostic(ErrorCode.ERR_MemberAlreadyExists, "M").WithArguments("M", "Base").WithLocation(5, 25)
+                );
+
+            var member = (SourceMethodSymbol)comp.GlobalNamespace.GetMember("Derived.M");
+            bool useMethodImpl = member.RequiresExplicitOverride(out bool shouldWarn);
+            Assert.False(shouldWarn);
+            Assert.Equal(withCovariantReturns, useMethodImpl);
+
+            verify(SourceView(comp, ""));
+            verify(CompilationReferenceView(comp, ""));
+            verify(RetargetingView(comp, ""));
+
+            void verify(CSharpCompilation compilation)
+            {
+                VerifyOverride(compilation,
+                    methodName: "Derived.M",
+                    overridingMemberDisplay: "void Derived.M(System.Int32 z)",
+                    overriddenMemberDisplay: "void Base.M(System.Int32 x)",
+                    requiresMethodimpl: useMethodImpl);
+            }
+        }
+
+        [Theory]
+        [CombinatorialData]
+        public void DuplicateDeclarations_02(
+            bool withCovariantReturns
+            )
+        {
+            var source = @"
+class Base<T>
+{
+    public virtual void M(int x) => throw null; // 1
+    public virtual void M(int y) => throw null; // 2
+}
+
+class Derived : Base<int>
+{
+    public override void M(int z) => throw null; // 3
+}
+";
+            var comp = CreateCompilation(withCovariantReturns: withCovariantReturns, source);
+            comp.VerifyDiagnostics(
+                // (5,25): error CS0111: Type 'Base<T>' already defines a member called 'M' with the same parameter types
+                //     public virtual void M(int y) => throw null; // 2
+                Diagnostic(ErrorCode.ERR_MemberAlreadyExists, "M").WithArguments("M", "Base<T>").WithLocation(5, 25),
+                // (10,26): error CS0462: The inherited members 'Base<T>.M(int)' and 'Base<T>.M(int)' have the same signature in type 'Derived', so they cannot be overridden
+                //     public override void M(int z) => throw null; // 3
+                Diagnostic(ErrorCode.ERR_AmbigOverride, "M").WithArguments("Base<T>.M(int)", "Base<T>.M(int)", "Derived").WithLocation(10, 26)
+                );
+
+            var member = (SourceMethodSymbol)comp.GlobalNamespace.GetMember("Derived.M");
+            bool useMethodImpl = member.RequiresExplicitOverride(out bool shouldWarn);
+            Assert.False(shouldWarn);
+            Assert.Equal(withCovariantReturns, useMethodImpl);
```

Comment:
>Assert.Equal(withCovariantReturns, useMethodImpl); [](start = 12, length = 50)

This should always be false

Reply:
For reasons previously explained, this value is intended and by design, and not affected by any changes in this PR.

---
In reply to: [452975955](https://github.com/dotnet/roslyn/pull/45849#discussion_r452975955) [](ancestors = 452975955)